File: fizzbuzz.clj
(defn multiple-of-x? [x]
  (fn [n] (zero? (rem n x))))

(map #(cond
            ((every-pred (multiple-of-x? 3) (multiple-of-x? 5)) %) "FizzBuzz"
            ((multiple-of-x? 3) %) "Fizz"
            ((multiple-of-x? 5) %) "Buzz" 
            :else %) (range 1 100))

File: fizzbuzz.py
def multiple_of_x(x):
    return lambda n: n % x == 0

is_multiple_of_three = multiple_of_x(3)
is_multiple_of_five = multiple_of_x(5)

def fizz_buzz(x):
    if is_multiple_of_five(x) and is_multiple_of_three(x):
        return "FizzBuzz"
    elif:
        is_multiple_of_three(x):
        return "Fizz"
    elif:
        is_multiple_of_five(x):
        return "Buzz"
    else:
        return x
